# This service contains the definition for a request to the motion
# planner and the output it provides

MotionPlanRequest motion_plan_request

---

# A solution trajectory, if found
arm_navigation_msgs/RobotTrajectory trajectory

# The corresponding robot state
arm_navigation_msgs/RobotState robot_state

# Planning time
duration planning_time

# Error code - encodes the overall reason for failure
arm_navigation_msgs/ArmNavigationErrorCodes error_code

# More detailed error codes (optional) - encode information about each point in the returned trajectory
arm_navigation_msgs/ArmNavigationErrorCodes[] trajectory_error_codes
